### PR TITLE
add jwt reflect and resource config to support jwt authentication type

### LINKED ISSUE #48: Add jwt support
While poking around with the gradle part I used mostly jwt instead of oauth. Eventually running the native image failed because of missing jjwt classes. Is that only related to gradle or do we only support oauth for now?

### PR DESCRIPTION
**Pull request by atomfrede**
Native image can be build with jwt support and started. I got some errors when I tried to save a user (entity) but I seemed unrelated. Will open another issue for that.

@alina-yur especially for you 

closes #48

### CODE REVIEW

**Comment by alina-yur** [comment]
this will be very helpful, thank you, @atomfrede!👏

**Comment by atomfrede** [comment]
Looks like the threadpool for sending mails async is not working properly (non webflux). Going to investigate today. Besides that it looks quite well.

**Comment by atomfrede** [comment]
Very strange, after I disabled async mail sending, the thymeleaf template can't be compile, because of an error with `org.thymeleaf.exceptions.TemplateProcessingException: Exception evaluating SpringEL expression: "locale.language"`. Which is confusing as we have the same template with webflux and I can see there is a locale (which has of course a public getLanguage method).

**Comment by atomfrede** [comment]
Seems I found a solution. Let me update the PR

**Comment by atomfrede** [comment]
@mraible There seems to be an issue with macos ci. Will restart the failed builds, but everything works fine, now also with JWT 🎉

File: test-integration/samples/postgresql-webflux-jwt.jdl [hunk 1]
```diff
@@ -0,0 +1,35 @@
+application {
+  config {
+    applicationType monolith
+    authenticationType oauth2
```
Comment (mshima) [inline test-integration/samples/postgresql-webflux-jwt.jdl:4]:
```suggestion
```

Comment (atomfrede) [reply test-integration/samples/postgresql-webflux-jwt.jdl:4]: Thanks good catch 🤦

**Review by mshima — APPROVED** [review]
macOS docker install is flaky. It should be ok to merge on install failures.
Samples are ok now, should be ok to merge if succeed.

**Comment by atomfrede** [comment]
I have fixed remaining issues with webflux and reflect config. Macos build is working fine locally, so as @mshima said flaky environment. I would say let's merge it and create a new release.

**Comment by atomfrede** [comment]
bounty claimed https://opencollective.com/generator-jhipster/expenses/91302

**Pull request merged**